### PR TITLE
Hw2 task05

### PR DESCRIPTION
**Pull request by mirmozavr**

### CODE REVIEW

File: hw2/task05/hw2_task05.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+it behaves as range function:
+import string
+assert = custom_range(string.ascii_lowercase, 'g') == ['a', 'b', 'c', 'd', 'e', 'f']
+assert = custom_range(string.ascii_lowercase, 'g', 'p') == ['g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o']
+assert = custom_range(string.ascii_lowercase, 'p', 'g', -2) == ['p', 'n', 'l', 'j', 'h']
+"""
+
+
+def custom_range(
+    it_obj: str, start: str = None, end: str = None, step: int = 1
```
Comment (Cjkjvfnby) [inline hw2/task05/hw2_task05.py]: Please follow the description, not by example.  `any iterable of unique values`  is not only str

File: hw2/task05/hw2_task05.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    if start is None:
+        start = 0
+    else:
+        start = it_obj.index(start)
+    if end is None:
+        end, start = start, 0
+    else:
+        end = it_obj.index(end)
+
+    for i in range(start, end, step):
```
Comment (Cjkjvfnby) [inline hw2/task05/hw2_task05.py]: How else could you slice iterable to get the required list?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

**Comment by Cjkjvfnby** [comment]
This PR shows me only partially changed files.   Looks like you have merged the first implementation to the master. Please fix it.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
This PR shows me changes in the homework files, but I expect full files to be present here.   Probably you have merged the first version to the master. Please fix it.

**Comment by mirmozavr** [comment]
@Cjkjvfnby I forgot to checkout from master to this branch at 1st iteration. But this branch do have all versions of task files. I plan to resolve conflicts with master branch after this branch approval. Right?

**Comment by Cjkjvfnby** [comment]
I expect that `hw2_task05.py ` present in this branch and not present in `master` branch. 

In this case, diff will show me the whole file.  Now I can see the difference between the version in this branch and the version in master.  See https://github.com/mirmozavr/EPAMHW/pull/10/files

**Comment by mirmozavr** [comment]
Removed files that belong to this branch from 'master'

**Comment by Cjkjvfnby** [comment]
> Removed files that belong to this branch from 'master'

Did it solve the initial problem in the https://github.com/mirmozavr/EPAMHW/pull/10/files ?

Looks like you need to merge master to this branch.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**